testing/src/test/java/io/stargate/it/cql/SystemTablesTest.java

```diff
@@ -1,29 +1,94 @@
 package io.stargate.it.cql;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
 import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.google.common.collect.Streams;
 import io.stargate.it.BaseOsgiIntegrationTest;
 import io.stargate.it.driver.CqlSessionExtension;
 import io.stargate.it.storage.StargateConnectionInfo;
 import io.stargate.it.storage.StargateEnvironmentInfo;
-import java.net.UnknownHostException;
+import io.stargate.it.storage.StargateSpec;
+import java.net.InetAddress;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
+@StargateSpec(nodes = 2)
 @ExtendWith(CqlSessionExtension.class)
-public class SystemTablesTest extends BaseOsgiIntegrationTest {
+class SystemTablesTest extends BaseOsgiIntegrationTest {
+  @Test
+  @DisplayName("Should expose Stargate addresses in system.local and system.peers")
+  public void querySystemLocalAndPeers(CqlSession session) {
+    Iterator<Node> nodes = session.getMetadata().getNodes().values().iterator();
+
+    Node localNode = nodes.next();
```

Comment:
Can we be sure that the first node in the list is the one to which we connected? Can we get the local node address from the session?

Reply:
I know that the `StargateContainer` extension always uses the first node for injecting connection info objects, but in this case the CQL session extension is also involved, so the association of the node and session is less direct :thinking:

Reply:
It's the local node because the test makes explicit: `SimpleStatement.builder("SELECT * FROM system.local").setNode(localNode)`.

Reply:
Ah... I missed that, TIL :)